cmd/skopeo/copy.go

```diff
@@ -99,23 +102,40 @@ See skopeo(1) section "IMAGE NAMES" for the expected format
 }
 
 // parseMultiArch parses the list processing selection
-// It returns the copy.ImageListSelection to use with image.Copy option
-func parseMultiArch(multiArch string) (copy.ImageListSelection, error) {
-	switch multiArch {
-	case "system":
-		return copy.CopySystemImage, nil
-	case "all":
-		return copy.CopyAllImages, nil
-	// There is no CopyNoImages value in copy.ImageListSelection, but because we
-	// don't provide an option to select a set of images to copy, we can use
-	// CopySpecificImages.
-	case "index-only":
-		return copy.CopySpecificImages, nil
-	// We don't expose CopySpecificImages other than index-only above, because
-	// we currently don't provide an option to choose the images to copy. That
-	// could be added in the future.
+// It returns the copy.ImageListSelection, instance list, and platform list to use in image.Copy option
+func parseMultiArch(multiArch string) (copy.ImageListSelection, []digest.Digest, []imgspecv1.Platform, error) {
+	switch {
+	case multiArch == "system":
+		return copy.CopySystemImage, nil, nil, nil
+	case multiArch == "all":
+		return copy.CopyAllImages, nil, nil, nil
+	case multiArch == "index-only":
+		// There is no CopyNoImages value in copy.ImageListSelection per se, but
+		// we can get the desired effect by using CopySpecificImages.
+		return copy.CopySpecificImages, nil, nil, nil
+	case strings.HasPrefix(multiArch, "sparse:"):
+		sparseArg := strings.TrimPrefix(multiArch, "sparse:")
+		platformSpecs := strings.Split(sparseArg, ",")
+		var platformList []imgspecv1.Platform
+		var instanceList []digest.Digest
+		for _, platformSpec := range platformSpecs {
+			if instanceDigest, err := digest.Parse(platformSpec); err == nil {
```

Comment:
Hum… I’m instinctively not a fan of “if it parses like A, it is A, else it is B” formats; OTOH it _can_ be a convenient UI.

(That… kinda applies to the `system` string, but that would be absurd.)

---

Maybe write some kind of an explicit rule that digests contain a `:`, but non-digests don’t contain it? I didn’t read all of the `platforms` package to see whether that would work (OTOH if it didn’t, we… might be in trouble here)